### PR TITLE
PT-187612052 filtered list of permission forms, and form for creating new permission forms

### PR DESCRIPTION
**Pull request by bacalj**
This creates a filterable list of permission forms, and a form for creating new ones, as described in PT-187612052

- update api routes and define create function in api
- remove unnecessary passing of api url through rails view
- update and create path constants for permission forms and projects
- use a resource route instead of namespace with specified for now
- component tree for index, permission form creation form, listing row
- styling
- creation of basic re-usable `useFetch` hook
- rspec api route smoke test

### CODE REVIEW

**Review by pjanik — APPROVED** [review]
Looks good to me 👍

File: rails/react-components/src/library/components/permission-forms-v2/use-fetch.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from "react";
```
Comment (dougmartin) [inline rails/react-components/src/library/components/permission-forms-v2/use-fetch.ts:1]: Maybe move this to a shared `hooks` folder under `src/library` (created folder if needed)?

**Review by dougmartin — APPROVED** [review]
Looks great :+1:

File: rails/spec/controllers/api/v1/permission_forms_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'spec_helper'
+
+RSpec.describe API::V1::PermissionFormsController, type: :controller do
+  describe '#index' do
+    it 'GET index' do
+      skip "Skipping this test for now"
```
Comment (dougmartin) [inline rails/spec/controllers/api/v1/permission_forms_controller_spec.rb:6]: You should probably go ahead and add the spec test before it gets left unimplemented.

**Review by pjanik — APPROVED** [review]
Still looks good to me 👍 

I left some minor inline comments. Nothing critical, so it doesn't need to block merging this PR.

File: rails/react-components/src/library/components/permission-forms-v2/use-fetch.ts (hunk 1 above)
Comment (bacalj) [reply rails/react-components/src/library/components/permission-forms-v2/use-fetch.ts:1]: Great, done, thanks.

File: rails/react-components/src/library/components/permission-forms-v2/index.tsx [hunk 3]
```diff
@@ -1,37 +1,91 @@
...
+  const [permissionForms, setPermissionForms] = useState<IPermissionForm[]>([]);
+  const [projects, setProjects] = useState<IProject[]>([]);
+
+  // State for UI
+  const [showForm, setShowForm] = useState(false);
+  const [currentSelectedProject, setCurrentSelectedProject] = useState<number | "">("");
+  const [visibleForms, setVisibleForms] = useState(permissionForms);
+
+  // Update state when data is fetched
+  useEffect(() => setPermissionForms(permissionsData), [permissionsData]);
```
[the remark this answers is not in the document]
Comment (bacalj) [reply rails/react-components/src/library/components/permission-forms-v2/index.tsx:24]: Ah thank you, I simplified as you described.  I was keeping that `permissionsForm` state just so I could update it when the new form was created.  But co-pilot informed me that I could add `refetch` function for the hook to return with the data.

File: rails/react-components/src/library/components/permission-forms-v2/index.tsx [hunk 4]
```diff
@@ -1,37 +1,91 @@
...
+
+  // update visible permissions list when project changes
+  useEffect(() => {
+    const belongsToSelectedProject = (form: IPermissionForm) => form.project_id === Number(currentSelectedProject);
+    const formsToDisplay = currentSelectedProject === ""
+    ? permissionForms
+    : permissionForms?.filter(belongsToSelectedProject);
+
+    setVisibleForms(formsToDisplay);
+  }, [permissionForms, currentSelectedProject]);
```
[the remark this answers is not in the document]
Comment (bacalj) [reply rails/react-components/src/library/components/permission-forms-v2/index.tsx:35]: Removed, thanks!

**Pull request merged**